client/service/src/builder.rs

```diff
@@ -781,12 +781,10 @@ where
 		return Err("Warp sync enabled, but no warp sync provider configured.".into())
 	}
 
-	if config.state_pruning.is_archive() {
-		match config.network.sync_mode {
-			SyncMode::Fast { .. } => return Err("Fast sync doesn't work for archive nodes".into()),
-			SyncMode::Warp => return Err("Warp sync doesn't work for archive nodes".into()),
-			SyncMode::Full => {},
-		};
+	if client.requires_full_sync() &&
+		!matches!(config.network.sync_mode, sc_network::config::SyncMode::Full)
+	{
+		return Err("The backend settings require the full-sync mode".into())
```

Comment:
```suggestion
	if client.requires_full_sync() {
		match config.network.sync_mode {
			SyncMode::Fast { .. } => return Err("Fast sync doesn't work for archive nodes".into()),
			SyncMode::Warp => return Err("Warp sync doesn't work for archive nodes".into()),
			SyncMode::Full => {},
		};
```

I think these error messages are little bit more clear.

Reply:
Those wordings are definitely more clear, but they just happen to tell the truth with the current implementation of `requires_full_sync(&self) -> bool`.

Perhaps the "question" to the client should be formulated as `is_archive_node(&self) -> bool` in that case...

(I couldn't expose the exact `PruningMode` via any of the `sc-client-api` interfaces because I didn't see it fit to add another dependency to that crate: either `sc-state-db` or `sc-client-db`).

Reply:
> Those wordings are definitely more clear, but they just happen to tell the truth with the current implementation of `requires_full_sync(&self) -> bool`.
> 
> Perhaps the "question" to the client should be formulated as `is_archive_node(&self) -> bool` in that case...

Yes that is being correct. I mean we could change the error message to not contain `for archive nodes` and replace that with something like "when the backed requires full sync mode". 

Or we expose the pruning mode to make it work like before.